lib/chef_zero/server.rb

```diff
@@ -564,54 +583,54 @@ def global_endpoint?(ep)
     end
 
     def app
-      return @app if @app
-      router = RestRouter.new(open_source_endpoints)
-      router.not_found = NotFoundEndpoint.new
+      @app ||= begin
+        router = RestRouter.new(open_source_endpoints)
+        router.not_found = NotFoundEndpoint.new
 
-      if options[:single_org]
-        rest_base_prefix = [ 'organizations', options[:single_org] ]
-      else
-        rest_base_prefix = []
-      end
-      @app = proc do |env|
-        begin
-          prefix = global_endpoint?(env['PATH_INFO']) ? [] : rest_base_prefix
+        if options[:single_org]
+          rest_base_prefix = [ 'organizations', options[:single_org] ]
+        else
+          rest_base_prefix = []
+        end
+        proc do |env|
+          begin
+            prefix = global_endpoint?(env['PATH_INFO']) ? [] : rest_base_prefix
 
-          request = RestRequest.new(env, prefix)
-          if @on_request_proc
-            @on_request_proc.call(request)
-          end
-          response = nil
-          if @request_handler
-            response = @request_handler.call(request)
-          end
-          unless response
-            response = router.call(request)
-          end
-          if @on_response_proc
-            @on_response_proc.call(request, response)
-          end
+            request = RestRequest.new(env, prefix)
+            if @on_request_proc
```

Comment:
What about some getters/setters for `@on_request_proc`, `@request_handler` and `@on_response_proc`?